src/hooks/useArray.ts

```diff
@@ -82,14 +90,27 @@ export const useArray = <T>(
     },
     [items]
   );
-
+  console.log(isDirty);
   const clear: UseArrayClear<T> = useCallback(() => {
     const result: T[] = [];
     setItems(result);
     setIsDirty(true);
     return result;
   }, [setItems]);
 
+  const replace: UseArrayReplace<T> = useCallback(
+    (newArray: T[]) => {
+      const result = options?.prepare
+        ? newArray?.map(options.prepare)
+        : newArray;
+      clearDirty();
+      setPrepared(result);
+
+      return result;
+    },
+    [clearDirty, options?.prepare]
+  );
```

Comment:
The idea of `useArray()` is to always wrap the same object, the same array, hence the `clear` function.

If there is a need of new wrapped array in a component somewhere, it can be dealt with there (a simple `[...arrayFromUseArray]` creates new reference for outside purpose.